### ISSUE TITLE
Add support for ignore pattern in MethodName cop

### ISSUE THREAD

**Issue by Eneroth3** [rubocop-hq/rubocop#6075]
First of all, thanks for such a lovely tool!

Generally I think Ruby methods should be in snake case. However, SketchUp's Ruby API has several interfaces that calls methods named in camel case, e.g. the [Tool interface](http://ruby.sketchup.com/Sketchup/Tool.html) and the various Observer interfaces.

It would be useful if this cop could ignore specific, pre-defined method names.

````yaml
Naming/MethodName:
  IgnoredPatterns:
    - '^\s*onSelectionBulkChange\s.*'
    - '^\s*onSelectionCleared\s.*'
````

Using RuboCop version 0.51.0.

**Comment by Eneroth3** [comment]
I'm still interested in this. I think anyone who develops SketchUp plugins and use rubocop could benefit from this.

**Comment by koic** [comment]
Thanks for the feature request based on a practical use case. I opened a PR #7318.

**Comment by Eneroth3** [comment]
I'm sorry to get back and diss my own idea after people have spent time implementing this, but I have rethought this. Listing regular expressions that matches the whole line and have some boilerplate for handling leading whitespaces are trailing arguments doesn't make a whole lot of sense.

Would it be possible to re-implement this so it simply just matches the method name?

This doesn't read as good as...

```yml
Naming/MethodName:
  IgnoredPatterns:
    - '^\s*enableVCB\?(\(.*\))?$'
    - '^\s*getExtents(\(.*\))?$'
    - '^\s*getInstructorContentDirectory(\(.*\))?$'
    - '^\s*getMenu(\(.*\))?$'
    - '^\s*onCancel(\(.*\))?$'
    - '^\s*onKeyDown(\(.*\))?$'
    - '^\s*onKeyUp(\(.*\))?$'
    - '^\s*onLButtonDoubleClick(\(.*\))?$'
    - '^\s*onLButtonDown(\(.*\))?$'
    - '^\s*onLButtonUp(\(.*\))?$'
    - '^\s*onMButtonDoubleClick(\(.*\))?$'
    - '^\s*onMButtonDown(\(.*\))?$'
    - '^\s*onMButtonUp(\(.*\))?$'
    - '^\s*onMouseEnter(\(.*\))?$'
    - '^\s*onMouseLeave(\(.*\))?$'
    - '^\s*onMouseMove(\(.*\))?$'
    - '^\s*onMouseWheel(\(.*\))?$'
    - '^\s*onRButtonDown(\(.*\))?$'
    - '^\s*onRButtonUp(\(.*\))?$'
    - '^\s*onReturn(\(.*\))?$'
    - '^\s*onSetCursor(\(.*\))?$'
    - '^\s*onUserText(\(.*\))?$'
```
...this.

```yml
Naming/MethodName:
  AllowedNames:
    - enableVCB?
    - getExtents
    - getInstructorContentDirectory
    - getMenu
    - onCancel
    - onKeyDown
    - onKeyUp
    - onLButtonDoubleClick
    - onLButtonDown
    - onLButtonUp
    - onMButtonDoubleClick
    - onMButtonDown
    - onMButtonUp
    - onMouseEnter
    - onMouseLeave
    - onMouseMove
    - onMouseWheel
    - onRButtonDown
    - onRButtonUp
    - onReturn
    - onSetCursor
    - onUserText
```

This time ´AllowedNames´ is chosen to be consistent with the `Naming/UncommunicativeMethodParamName` cop.

Again, sorry for not thinking this through in the first place.

**Comment by Renderiza** [comment]
> I'm sorry to get back and diss my own idea after people have spent time implementing this, but I have rethought this. Listing regular expressions that matches the whole line and have some boilerplate for handling leading whitespaces are trailing arguments doesn't make a whole lot of sense.
> 
> Would it be possible to re-implement this so it simply just matches the method name?
> 
> This doesn't read as good as...
> 
> ```yaml
> Naming/MethodName:
>   IgnoredPatterns:
>     - '^\s*enableVCB\?(\(.*\))?$'
>     - '^\s*getExtents(\(.*\))?$'
>     - '^\s*getInstructorContentDirectory(\(.*\))?$'
>     - '^\s*getMenu(\(.*\))?$'
>     - '^\s*onCancel(\(.*\))?$'
>     - '^\s*onKeyDown(\(.*\))?$'
>     - '^\s*onKeyUp(\(.*\))?$'
>     - '^\s*onLButtonDoubleClick(\(.*\))?$'
>     - '^\s*onLButtonDown(\(.*\))?$'
>     - '^\s*onLButtonUp(\(.*\))?$'
>     - '^\s*onMButtonDoubleClick(\(.*\))?$'
>     - '^\s*onMButtonDown(\(.*\))?$'
>     - '^\s*onMButtonUp(\(.*\))?$'
>     - '^\s*onMouseEnter(\(.*\))?$'
>     - '^\s*onMouseLeave(\(.*\))?$'
>     - '^\s*onMouseMove(\(.*\))?$'
>     - '^\s*onMouseWheel(\(.*\))?$'
>     - '^\s*onRButtonDown(\(.*\))?$'
>     - '^\s*onRButtonUp(\(.*\))?$'
>     - '^\s*onReturn(\(.*\))?$'
>     - '^\s*onSetCursor(\(.*\))?$'
>     - '^\s*onUserText(\(.*\))?$'
> ```
> 
> ...this.
> 
> ```yaml
> Naming/MethodName:
>   AllowedNames:
>     - enableVCB?
>     - getExtents
>     - getInstructorContentDirectory
>     - getMenu
>     - onCancel
>     - onKeyDown
>     - onKeyUp
>     - onLButtonDoubleClick
>     - onLButtonDown
>     - onLButtonUp
>     - onMButtonDoubleClick
>     - onMButtonDown
>     - onMButtonUp
>     - onMouseEnter
>     - onMouseLeave
>     - onMouseMove
>     - onMouseWheel
>     - onRButtonDown
>     - onRButtonUp
>     - onReturn
>     - onSetCursor
>     - onUserText
> ```
> 
> This time ´AllowedNames´ is chosen to be consistent with the `Naming/UncommunicativeMethodParamName` cop.
> 
> Again, sorry for not thinking this through in the first place.

Configured '.robocop.yml' with your suggestion and it works great while developing SketchUp Extensions. Thanks!
